src/pfe/file-watcher/server/src/projects/OdoExtensionProject.ts

```diff
@@ -0,0 +1,275 @@
+        projectUtil.containerCreate(operation, path.join(this.fullPath, "odo-extension-entrypoint.sh"), "create");
+    }
+
+    /**
+     * @function
+     * @description Update operation for the project.
+     *
+     * @param operation <Required | Operation> - The update operation.
+     * @param changedFiles <Optional | projectEventsController.IFileChangeEvent[]> - The file changed event array.
+     */
+    update = (operation: Operation, changedFiles?: projectEventsController.IFileChangeEvent[]): void => {
+        projectUtil.containerUpdate(operation, path.join(this.fullPath, "odo-extension-entrypoint.sh"), "update");
+    }
+
+    /**
+     * @function
+     * @description Rebuild operation for the project.
+     *
+     * @param projectInfo <Required | ProjectInfo> - The metadata information for the project.
+     */
+    rebuild = async (projectInfo: ProjectInfo): Promise<void> => {
+        await projectUtil.runScript(projectInfo, path.join(this.fullPath, "odo-extension-entrypoint.sh"), "rebuild");
+    }
+
+    /**
+     * @function
+     * @description Delete operation for the project.
+     *
+     * @param projectInfo <Required | ProjectInfo> - The metadata information for the project.
+     */
+    deleteContainer = async (projectInfo: ProjectInfo): Promise<void> => {
+        await projectUtil.containerDelete(projectInfo, path.join(this.fullPath, "odo-extension-entrypoint.sh"));
+    }
+
+    /**
+     * @function
+     * @description Validate project.
+     *
+     * @param operation <Required | Operation> - The validate operation for the project.
+     */
+    validate = async (operation: Operation): Promise<void> => {
+        const validator = new Validator(operation);
+        await validator.validateRequiredFiles(this.config.requiredFiles);
+        validator.sendResult();
+    }
+
+    /**
+     * @function
+     * @description Get the build log path for the project.
+     *
+     * @param logDirectory <Required | string> - The log location directory.
+     *
+     * @returns Promise<BuildLog>
+     */
+    getBuildLog = async (logDirectory: string): Promise<BuildLog> => {
+        return await logHelper.getBuildLogs(logDirectory, this.config.buildLogs);
+    }
+
+    /**
+     * @function
+     * @description Get the app log path for the project.
+     *
+     * @param logDirectory <Required | String> - The log location directory.
+     *
+     * @returns Promise<AppLog>
+     */
+    getAppLog = async (logDirectory: string): Promise<AppLog> => {
+        return await logHelper.getAppLogs(logDirectory, this.config.appLogs);
+    }
+
+    /**
+     * @function
+     * @description Get the project capabilities of the project.
+     *
+     * @returns ProjectCapabilities
+     */
+    getCapabilities = (): ProjectCapabilities => {
+
+        if (this.config.capabilities) {
+            return new ProjectCapabilities(this.config.capabilities.startModes, this.config.capabilities.controlCommands);
```

Comment:
I think the default capability supports startModes of debug and debugNoInit. Given that odo does not support debug, should we filter those out?

Reply:
We already filtered those out in ODO extension implementation here: https://github.com/eclipse/codewind-odo-extension/blob/master/.odo-extension-config.json#L9, then we load the capability in Turbine layer.